### PR TITLE
pkg: iotivity: Hide put_light function if not needed

### PR DESCRIPTION
**Pull request by rzr**
As requested in:
https://github.com/RIOT-OS/RIOT/pull/6075#discussion_r96491505

Bug: https://github.com/RIOT-OS/RIOT/issues/6241
Change-Id: I6b2af5ab6b8bf20e848dcd30d580990006c950e7
Signed-off-by: Philippe Coval <philippe.coval@osg.samsung.com>

### CODE REVIEW

File: examples/iotivity_examples/client_switch/client_switch.c [hunk 1]
```diff
@@ -76,6 +76,7 @@ stop_observe(void *data)
     return DONE;
 }
 
+#ifdef BUTTON_GPIO
```
Comment (aabadie) [inline examples/iotivity_examples/client_switch/client_switch.c]: I may have missed something but it's totally unclear to me why this is required : the code of this function has nothing to do with the board definitions.

**Comment by rzr** [comment]
compiler was complaining about unused function

**Comment by aabadie** [comment]
Ok, got it.

**Comment by kaspar030** [comment]
Please be aware that this is not compiled by our CI's.

**Comment by aabadie** [comment]
Ok, so I guess it can be merged like this as 1 of the 2 CIs is green.

**Comment by rzr** [comment]
There is no emergency, as long as iotivity is not evicted I can wait ;)

**Comment by kaspar030** [comment]
> here is no emergency, as long as iotivity is not evicted I can wait ;)

We don't want to evict iotivity, just make sure it meets our quality standards.
Unfortunately, running "make buildtest" in examples/iotivity/client_switch fails for *all* platforms.
We usually don't accept that, even if ```use CFLAGS=-Wno-error``` like hints make code compile for some platforms. iotivity got lucky as the initial reviewers where not aware that third level folders in examples/ don't get compiled.

**Comment by rzr** [comment]
I hope original comitters are aware, I'll try to help on fixing this anyway it's a good exercise to learn about RIOT project, are any of you at FOSDEM?

**Comment by emmanuelsearch** [comment]
there might be one of us at FOSDEM. Not certain yet, though.

**Comment by emmanuelsearch** [comment]
No changes in the sense that I tried make buildtest with #6403 and with the latest commit of iotivity-constrained (aed3afb26107a1e32c9891b425f8ae2563b19d6e) and all builds fail...

**Comment by kYc0o** [comment]
All green here, how do we proceed? @kaspar030 ?

**Comment by kYc0o** [comment]
It has no effect only for Murdock or also for Jenkins?

**Comment by kaspar030** [comment]
No effect for both.

**Comment by aabadie** [comment]
Then let's merge !

**Pull request merged**